### PR TITLE
🐛 amp-carousel: hide next/prev in type=carousel for screen readers

### PR DESCRIPTION
**Pull request by aghassemi**
Fixes https://github.com/ampproject/amphtml/issues/21071

In scrollable carousel, the next/previous buttons add no functionality for screen readers as scrollable carousel is just a horizontally scrollable div which ATs navigate just like any other content.

To avoid confusion, we therefore set the role to presentation for the controls in this case.

### CODE REVIEW

**Comment by sparhami** [comment]
My understanding is that this is intentional for `type="carousel"`. The next/prev buttons don't actually go to the next/previous slide, but instead move the scroll position by 1 carousel viewport forwards/backwards. Unlike `type="slides"`, other slides are not removed from the a11y tree (using `aria-hidden`), so as far as a screen reader user is concerned, these buttons don't actually go to the next/previous slide. They still bring the slide into the viewport, which causes them to do layout, but I don't think adding a label as if they go to the next/prev slide is correct.

For the new version of carousel (when configured to be the equivalent of `type="carousel"`), my plan is to make it the carousel/slides behave as a list (give each item role="listitem") and make the next/previous buttons `aria-hidden="true"`. The first part is done (make it into a list), but the second part still needs to be done.

The screen reader should bring the list items into view as you navigate them (need to verify that this is correct, may need to add tabindex to get this behavior), which should cause them to do layout.

**Comment by aghassemi** [comment]
@sparhami I will change the approach to hide those buttons via `role=presentation` then, no point in having buttons that show up in controls list in this case.

For `amp-stream-gallery`, using `list` sounds great but may not be enough; given these horizontal areas are sort of secondary content, it should be super easy for AT users to skip them without going through every item, same way a visual user can skim and move on. 

I am thinking we may need to have a `skip link` every 3-5 slides. List navigation with desktop ATs probably has good commands to jump to last item/out of the list but I could not find anything for VoiceOver/TalkBack.

WDYT about about skip links? Maybe something like `skip the next <remaining> items` as the text.

File: extensions/amp-carousel/0.1/base-carousel.js [hunk 1]
```diff
@@ -88,6 +88,16 @@ export class BaseCarousel extends AMP.BaseElement {
     return button;
   }
 
+  /**
+   * The ARIA role for the controls. Either `button` or `presentation` based
+   * on usage.
+   * @return {string}
```
Comment (sparhami) [inline extensions/amp-carousel/0.1/base-carousel.js]: `@protected`

**Comment by Jessman5** [comment]
+1 for skiplinks. I'm currently implementing them for general navigation (header, navigation, search, main, footer, etc) in my project. They always come in handy.

**Pull request merged**